apps/ejabberd/src/mod_mam_utils.erl

```diff
@@ -227,38 +233,51 @@ external_binary_to_mess_id(BExtMessID) when is_binary(BExtMessID) ->
 %% -----------------------------------------------------------------------
 %% XML
 
+%% @doc Adds all arcid elements (for backward compatibility)
```

Comment:
So far the 'archived' element was added only if Mongoose was configured to add it (by default disabled). How about we add 'stanza-id' always (as this will not break compatibility) and add 'archived' only if configured?